From 0682141c96f73c5eb19eedcc75ade108e578df64 Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Fri, 31 Dec 2021 08:24:32 +0200
Subject: [PATCH] [SW-36155] rma: use api instead of global

Add api for i2csi_set_rma_failure instead of
global, which is also needed as preparation
for using i2c-indirect device headers.

Change-Id: I1027ee61098b08e3592259f01b3936aabcd4a120
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/arch/gaudi/arm/rma_info.c b/arch/gaudi/arm/rma_info.c
index bda1caa..b6cc63a 100644
--- a/arch/gaudi/arm/rma_info.c
+++ b/arch/gaudi/arm/rma_info.c
@@ -215,7 +215,7 @@
 		cpu_boot_err_set(CPU_BOOT_ERR_DEVICE_UNUSABLE_FAIL);
 		preboot_puts_raw("RMA Fatal error(s): ");
 		preboot_rma_type_print(rma_header->rma_failure);
-		i2c_rma_failure = rma_header->rma_failure;
+		i2csi_set_rma_failure(&rma_header->rma_failure);
 		if (rma_header->rma_failure.hbm_init)
 			rma_hbm_init_err = rma_header->hbm_init_err;
 	}
diff --git a/include/i2cslave_indirect.h b/include/i2cslave_indirect.h
index 6074ccf..27124a2 100644
--- a/include/i2cslave_indirect.h
+++ b/include/i2cslave_indirect.h
@@ -27,5 +27,5 @@
 			bool is_linux_boot_fail);
 int i2csi_set_wd_err(enum hl_fw_reset_cause err);
 void i2csi_update_smbus_alert(struct rma_entry *rma_header);
-extern struct rma_failure i2c_rma_failure;
+void i2csi_set_rma_failure(struct rma_failure *rma_f);
 #endif /* _I2CSLAVE_INDIRECT_H_ */
diff --git a/lib/preboot/i2cslave_indirect.c b/lib/preboot/i2cslave_indirect.c
index bfea488..f5ee739 100644
--- a/lib/preboot/i2cslave_indirect.c
+++ b/lib/preboot/i2cslave_indirect.c
@@ -1179,3 +1179,8 @@
 
 	return 1;
 }
+
+void i2csi_set_rma_failure(struct rma_failure *rma_f)
+{
+	memcpy(&i2c_rma_failure, rma_f, sizeof(i2c_rma_failure));
+}
